src/plumpy/processes.py

```diff
@@ -1236,19 +1244,27 @@ def kill(self, msg_text: Optional[str] = None) -> Union[bool, asyncio.Future]:
             return False
 
         if self._killing:
-            # Already killing
-            return self._killing
+            self._killing.cancel()
+
+        if force_kill:
+            msg = MessageBuilder.kill(msg_text, force_kill=force_kill)
+            new_state = self._create_state_instance(process_states.ProcessState.KILLED, msg=msg)
+            self.transition_to(new_state)
+
+            # we cancel all existing tasks in the loop that are associated to this process
+            self.loop.create_task(self.cancel_all_tasks())
```

Comment:
As you can see, I removed the sending of an KillInterrupt because it always opens the possibility that something gets stuck when one wants to properly handle the release/cancellation of any resources (like the job on slurm). A force kill is not designed to have any kind of handling to free the resources that could be blocking. I therefore cancel all tasks in the loop so they cannot further block the worker.

I think the better solution for immediate but safe killing is to send another kill command with adapted EBM parameters (smaller interval). To allow that I implemented that if a new killing command is send, the old one is cancelled, so if the user sends multiple kills only the most recent one is active.

Reply:
> I therefore cancel all tasks in the loop so they cannot further block the worker.

I guess this will (onyl) work if the workchain is run not through submit to other daemon, which might be the initial goal where the `ctrl+C` to send the kill signal not work?

Reply:
My understanding was 1 worker <-> 1 process <-> 1 event loop, the worker will cancel everything related to this process. Anything that spawned this process (e.g. a workchain that is on a different worker) this will not affect and should also not. I am not sure about a scenario with two daemons, do we have this?

In any case I removed this since it can be blocking in case the remaining tasks in the event loop are blocking.